share/completions/git.fish

```diff
@@ -1463,8 +1475,21 @@ complete -k -f -c git -n '__fish_git_using_command switch; and not contains -- -
 complete -f -c git -n '__fish_git_using_command switch' -s c -l create -d 'Create a new branch'
 complete -f -c git -n '__fish_git_using_command switch' -s C -l force-create -d 'Force create a new branch'
 complete -f -c git -n '__fish_git_using_command switch' -s d -l detach -d 'Switch to a commit for inspection and discardable experiment'
+complete -f -c git -n '__fish_git_using_command switch' -l guess -d 'Guess branch name from remote branch (default)'
 complete -f -c git -n '__fish_git_using_command switch' -l no-guess -d 'Do not guess branch name from remote branch'
+complete -f -c git -n '__fish_git_using_command switch' -s f -l force -l discard-changes -d 'Proceed even if the index or the working tree differs from HEAD'
+complete -f -c git -n '__fish_git_using_command switch' -s m -l merge -d 'Merge the current branch and contents of the working tree into a new branch'
+complete -f -c git -n '__fish_git_using_command switch' -l 'conflict=merge' -d 'Same as --merge, but specify \'merge\' as the style of the conflicting hunks (default)'
```

Comment:
Note: `__fish_git_using_command` can take multiple arguments, so for the common options you can use one call with `__fish_git_using_command switch restore`.